### PR TITLE
Fix regexp matcher for special assets for the browser

### LINKED ISSUE #7077: File access permission question
## Expected Behavior
I have three pictures at https://play.minio.io:9000/xiaomaiyun/, favicon.ico and xiaomaigou_favicon.ico and xiaomaigoulogo.png, my buckets 'xiaomaiyun' didn't set any prefix, but when I was at When accessing an unauthenticated browser, the following occurs: When accessing https://play.minio.io:9000/xiaomaiyun/favicon.ico, normal access; when visiting https://play.minio.io: 9000/xiaomaiyun/xiaomaigou_favicon.ico, 500 Internal Server Error, and when visiting https://play.minio.io:9000/xiaomaiyun/xiaomaigoulogo.png, login is required, why? Why not all require login?

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used (`minio version`):
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

### PR DESCRIPTION
**Pull request by donatello**
Fixes #7077

## Regression

No.

## How Has This Been Tested?

Manually.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by donatello** [comment]
PR is updated with a fix for accessing URLs with a trailing slash - thanks to @kaankabalak for help with this!

**Review by kaankabalak — DISMISSED** [review]

File: cmd/web-router.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func assetFS() *assetfs.AssetFS {
 }
 
 // specialAssets are files which are unique files not embedded inside index_bundle.js.
-const specialAssets = ".*index_bundle.*.js$|.*loader.css$|.*logo.svg$|.*firefox.png$|.*safari.png$|.*chrome.png$|.*favicon.ico$"
+const specialAssets = "index_bundle.*.js|loader.css|logo.svg|firefox.png|safari.png|chrome.png|favicon.ico"
```
Comment (krishnasrinivas) [inline cmd/web-router.go:59]: can you check if the following simplification does not break anything:
```
diff --git a/cmd/web-router.go b/cmd/web-router.go
index 20067e0b..25f1de8c 100644
--- a/cmd/web-router.go
+++ b/cmd/web-router.go
@@ -19,7 +19,6 @@ package cmd
 import (
        "fmt"
        "net/http"
-       "strings"
 
        "github.com/elazarl/go-bindata-assetfs"
        "github.com/gorilla/handlers"
@@ -94,16 +93,9 @@ func registerWebRouter(router *mux.Router) error {
 
        // Add compression for assets.
        h := http.FileServer(assetFS())
-       compressedAssets := handlers.CompressHandler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-               idx := strings.LastIndex(r.URL.Path, slashSeparator)
-               if idx != -1 {
-                       r.URL.Path = r.URL.Path[idx+1:]
-               }
-               h.ServeHTTP(w, r)
-       }))
 
        // Serve javascript files and favicon from assets.
-       webBrowserRouter.Path(fmt.Sprintf("/{assets:%s}", specialAssets)).Handler(compressedAssets)
+       webBrowserRouter.Path(fmt.Sprintf("/{assets:%s}", specialAssets)).Handler(handlers.CompressHandler(h))
 
        // Serve index.html for rest of the requests.
        webBrowserRouter.Path("/{index:.*}").Handler(indexHandler{http.StripPrefix(minioReservedBucketPath, h)})
```

Comment (donatello) [reply cmd/web-router.go:59]: @krishnasrinivas I tried this, it doesnt work. We need to strip out the `/minio` prefix for the assetFS - but I got what you are trying to do. I am updating the PR with a simplification.

**Comment by donatello** [comment]
PR is updated, please review @kaankabalak @krishnasrinivas

**Review by krishnasrinivas — APPROVED** [review]

**Review by kaankabalak — APPROVED** [review]

**Pull request merged**